File: ModSim/Lista 1/Java/FallingParticle.java
public class FallingParticle extends Particle
{
  final static double g = 9.8;
  private double y0 = 0, v0 = 0;
  
  public FallingParticle(double y, double v)
  {
    System.out.println("A new FallingParticle object was created.");
    this.y = y;
    this.v = v;
    y0 = y;
    v0 = v;
  }
  
  public void step()
  {
    //System.out.println("blub");
    v = v - g * dt;
    y = y + v * dt;
    t = t + dt;
  }
  
  public double analyticPosition()
  {
    return y0 + v0 * t - (g * t * t) / 2.0;
  }
  
  public double analyticVelocity()
  {
    return v0 - g * t;
  }
}

File: ModSim/Lista 1/Python/FallingParticle.py
from Particle import Particle

class FallingParticle(Particle):
  g = 9.8
  __y0 = 0
  __v0 = 0
  
  def __init__ (self, y, v):
    print "A new FallingParticle object was created."
    self.y = y
    self.v = v
    self.__y0 = y
    self.__v0 = v
  
  def step(self):
    #print "fallingParticle method"
    self.v -= self.g * self.dt
    self.y += self.v * self.dt
    self.t += self.dt
  
  def analyticPosition(self):
    return self.__y0 + self.__v0 * self.t - (self.g * self.t * self.t) / 2.0
  
  def analyticVelocity(self):
    return self.__v0 - self.g * self.t
